situp-core/src/main/java/com/amazon/situp/parser/model/PipelineConfiguration.java

```diff
@@ -40,31 +41,41 @@ public PipelineConfiguration(
         this.readBatchDelay = getReadBatchDelayFromConfiguration(delay);
     }
 
-    private PluginSetting getSourceFromConfiguration(final Map<String, Map<String, Object>> sourceConfiguration) {
-        final PluginSetting sourcePluginSetting = getFirstPluginSettingFromConfiguration(sourceConfiguration,
-                SOURCE_COMPONENT);
-        if (sourcePluginSetting == null) {
+    private PluginSetting getSourceFromConfiguration(final Map.Entry<String, Map<String, Object>> sourceConfiguration) {
```

Comment:
You need to sync ur forked master, some of these changes are already in current master